project_stock_product_set/wizard/product_set_add.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2022 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+from odoo import fields, models
+
+
+class ProductSetAddFromTask(models.TransientModel):
+    _inherit = "product.set.add"
+    _name = "product.set.add.from.task"
+    _description = "product.set.add.from.task"
+
+    order_id = fields.Many2one(required=False)
+    task_id = fields.Many2one(
+        comodel_name="project.task",
+        string="Task",
+        required=True,
+        default=lambda self: self.env.context.get("active_id")
+        if self.env.context.get("active_model") == "project.task"
+        else None,
+        ondelete="cascade",
+    )
+
+    def _prepare_stock_move_lines(self):
+        move_lines = []
+        sm_model = self.env["stock.move"]
+        for _seq, set_line in enumerate(self._get_lines(), start=1):
+            values = set_line.prepare_stock_move_values(self.task_id, self.quantity)
+            values.update(sm_model.play_onchanges(values, values.keys()))
```

Comment:
But these lines should be done in the `prepare_stock_move_values` (and it should be `_prepare_stock_move_values`).